File: lib/models/timetable_entry.dart
class TimetableEntry {
  final int? id;
  final String subject;
  final String teacher;
  final String className;
  final String academicYear;
  final String dayOfWeek;
  final String startTime;
  final String endTime;
  final String room;
  final String? remoteId;

  TimetableEntry({
    this.id,
    required this.subject,
    required this.teacher,
    required this.className,
    required this.academicYear,
    required this.dayOfWeek,
    required this.startTime,
    required this.endTime,
    required this.room,
    this.remoteId,
  });

  factory TimetableEntry.fromMap(Map<String, dynamic> map) {
    final rawId = map['id'];
    return TimetableEntry(
      id: rawId is int ? rawId : null,
      subject: map['subject'] ?? '',
      teacher: map['teacher'] ?? '',
      className: map['className'] ?? '',
      academicYear: map['academicYear'] ?? '',
      dayOfWeek: map['dayOfWeek'] ?? '',
      startTime: map['startTime'] ?? '',
      endTime: map['endTime'] ?? '',
      room: map['room'] ?? '',
      remoteId: rawId is String ? rawId : map['remote_id'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      if (id != null) 'id': id,
      'subject': subject,
      'teacher': teacher,
      'className': className,
      'academicYear': academicYear,
      'dayOfWeek': dayOfWeek,
      'startTime': startTime,
      'endTime': endTime,
      'room': room,
      'remote_id': remoteId,
    };
  }
}

File: backend/src/Modules/Academics/SchoolManager.Modules.Academics/TimetableEntry.cs
using SchoolManager.BuildingBlocks.Multitenancy;

namespace SchoolManager.Modules.Academics;

public sealed class TimetableEntry : ITenantEntity
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string TenantId { get; set; } = string.Empty;
    public string Subject { get; set; } = string.Empty;
    public string Teacher { get; set; } = string.Empty;
    public string ClassName { get; set; } = string.Empty;
    public string AcademicYear { get; set; } = string.Empty;
    public string DayOfWeek { get; set; } = string.Empty;
    public string StartTime { get; set; } = string.Empty;
    public string EndTime { get; set; } = string.Empty;
    public string? Room { get; set; }

    public DateTimeOffset CreatedAtUtc { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAtUtc { get; set; } = DateTimeOffset.UtcNow;
}
